services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -4957,4 +4959,29 @@ public List<StoragePersona> getClusterStoragePersonas(String clusterName) {
   public List<String> cleanupInstanceCustomizedStates(String clusterName) {
     throw new VeniceUnsupportedOperationException("cleanupInstanceCustomizedStates");
   }
+
+  @Override
+  public StoreGraveyard getStoreGraveyard() {
+    return getVeniceHelixAdmin().getStoreGraveyard();
+  }
+
+  @Override
+  public void removeStoreFromGraveyard(String clusterName, String storeName) {
+    // Check parent data center to make sure store could be removed from graveyard.
+    getVeniceHelixAdmin().checkKafkaTopicAndHelixResource(clusterName, storeName, true, false, false);
+    // Check all child data centers to make sure store is removed from graveyard there.
+    Map<String, ControllerClient> controllerClientMap = getVeniceHelixAdmin().getControllerClientMap(clusterName);
+    controllerClientMap.forEach((coloName, cc) -> {
+      ControllerResponse response = cc.removeStoreFromGraveyard(storeName);
+      if (response.isError()) {
+        if (ErrorType.PRECONDITION_CHECK_FAILED.equals(response.getErrorType())) {
+          throw new PreconditionCheckFailedException(
```

Comment:
This error message should print out the reason about which precondition check failed

Reply:
If child controller responds error, parent controller will log the failure with error details:

2022-09-21 15:56:29 INFO [AdminSparkServer] [qtp377803945-811] [AUDIT] POST http://shuhliu-mn2.linkedin.biz:53325/remove_store_from_graveyard {cluster_name=venice-cluster0, hostname=localhost, store_name=testStoreGraveyardCleanupBatch_bab77ffea535_ff3c538f}
2022-09-21 15:56:29 INFO [AdminSparkServer] [qtp377803945-811] [AUDIT] FAILURE:  {"cluster":"venice-cluster0","name":"testStoreGraveyardCleanupBatch_bab77ffea535_ff3c538f","error":"Topic: venice_system_store_meta_store_testStoreGraveyardCleanupBatch_bab77ffea535_ff3c538f_rt still exists for store: testStoreGraveyardCleanupBatch_bab77ffea535_ff3c538f, please make sure all topics are removed.","errorType":"PRECONDITION_CHECK_FAILED","exceptionType":"BAD_REQUEST"} POST http://shuhliu-mn2.linkedin.biz:53325/remove_store_from_graveyard {cluster_name=venice-cluster0, hostname=localhost, store_name=testStoreGraveyardCleanupBatch_bab77ffea535_ff3c538f}

IMHO, parent controller does not need to log here again.

Reply:
SGTM. I was thinking from the point of an operator running the command - but if this is a periodic thread in parent controller, then it is okay to only log the colo's error